Automattic/wp-calypso
# Jetpack Error UX: Show error banner in Themes showcase

See https://github.com/Automattic/dotcom-forge/issues/3394

## Proposed Changes

Shows an error notice on the Themes showcase when Jetpack is unable to connect to the site:

<img width="1950" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/4a5f3de8-a8e6-4211-acc1-ce1924b47970">

Triggering logic was introduced in https://github.com/Automattic/wp-calypso/pull/79965

## Testing Instructions

1. Create a new Business site and take it Atomic.
2. Add some PHP code to `~/htdocs/wp-content/mu-plugins/local.php` that causes a fatal error.
3. Navigate to Themes in Calypso.
4. Verify the notice appears as expected.
5. Fix your fatal error.
6. Reload the Themes view.
7. Verify the notice no longer appears.

Flagging @Automattic/t-rex and @Automattic/lego for review in case there are other aspects that need to be tested/considered.
